### DISCUSSION TITLE
1.9b1

### DISCUSSION THREAD

**Discussion by sgillies** [Toblerity/Fiona#1171] [Announcements]
The fist 1.9 beta release is ready for early testing. All of the features planned for 1.9.0 are in this release.

The binary wheels on PyPI include GDAL 3.5.3, GEOS 3.11.1, and PROJ 9.0.1. To try the wheels, run `pip install --pre fiona`.

#### New features

* Add listdir and listlayers method to io.MemoryFile (resolving #754).
* Add support for TIN and triangle geometries (#1163).
* Add an allow_unsupported_drivers option to fiona.open() (#1126).
* Added support for the OGR StringList field type (#1141).

#### Changes and bug fixes

* Missing and unused imports have been added or removed.
* Make sure that errors aren't lost when a collection can't be saved properly (#1169).
* Ensure that ZipMemoryFile have the proper GDAL name after creation so that we can use listdir() (#1092).
* The fiona._loading module, which supports DLL loading on Windows, has been moved into __init__.py and is no longer used anywhere else (#1168).
* Move project metadata to pyproject.toml (#1165).
* Update drvsupport.py to reflect new format capabilities in GDAL 3.6.0 (#1122).
* Remove debug logging from env and _env modules.

<hr /><em>This discussion was created from the release <a href='https://github.com/Toblerity/Fiona/releases/tag/1.9b1'>1.9b1</a>.</em>

**Comment by jorisvandenbossche** [comment]
I was looking into the `fiona.model` related deprecation warnings that are introduced in Fiona 1.9 in context of geopandas' read_file/to_file functions: https://github.com/geopandas/geopandas/pull/2696

That PR updates geopandas' code to be compatible with future fiona (based on the warnings), but it made me wonder: is this needed for this case of passing data to fiona? 
I understand that _accessing_ the features/geometries as dict is deprecated, so fiona can expose a custom object (avoiding a direct conversion to a python dict). But that seems mostly important for _reading_? When writing data (passing features/geometries to fiona), fiona _could_ still provide the user convenience to be able to pass a dictionary, instead of having to convert this to a Feature/Geometry object manually? 

To be clear: it's a small and easy change in geopandas, so no problem at all for geopandas if it is decided that Fiona wants to be strict and only accept its own Feature/Geometry object.  
But for a direct user of fiona, it could be nice to still support the `__geo_interface__`-like objects out of the box? (https://gist.github.com/sgillies/2217756)

**Reply by sgillies** [reply]
@jorisvandenbossche I'm taking a look at it. Accepting `__geo_interface__` providers makes sense and shouldn't be hard to do.
